dcore/linear_solvers.py

```diff
@@ -0,0 +1,171 @@
+from firedrake import split, LinearVariationalProblem, \
+    LinearVariationalSolver, FunctionSpace, TestFunctions, TrialFunctions, \
+    TestFunction, TrialFunction, lhs, rhs, DirichletBC, FacetNormal, \
+    div, dx, jump, avg, dS_v, dS_h, inner
+
+class TimesteppingSolver(object):
+    """
+    Base class for timestepping linear solvers for dcore.
+
+    This is a dummy base class where the input is just copied to the output.
+
+    :arg x_in: :class:`.Function` object for the input
+    :arg x_out: :class:`.Function` object for the output
+    """
+
+    def __init__(x_in, x_out):
+        self.x_in = x_in #get input vector
+        self.x_out = x_out #get output vector
+    
+    def solve():
+        """
+        Function to execute the solver.
+        """
+        #This is a base class so we just copy x_in to x_out
+        self.x_out.assign(x_in)
```

Comment:
Huh, I don't know what you mean.  You can certainly add a docstring.  You can even have a method body which can be called with `super`.  You're just required to override this function in all direct subclasses of `TimeStepper`.